java/memory/src/main/java/org/apache/arrow/memory/DefaultAllocationManagerOption.java

```diff
@@ -85,14 +87,32 @@ static AllocationManagerType getDefaultAllocationManagerType() {
 
     switch (type) {
       case Netty:
```

Comment:
Let's do a follow-on where we change this from type to class name for the factory. That way we don't have weird strings in here. The implementations can all have the default factory class.